src/Evolution/Particles/MonteCarlo/CouplingTermsForPropagation.hpp

```diff
@@ -0,0 +1,37 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
```

Comment:
- include only `DataStructures/Tensor/TypeAliases.hpp` and `IndexType.hpp`
- forward declare not_null, DataVector, Packet
- move the includes to the cpp file